server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -293,7 +269,7 @@ private void onSuccessfulSegmentsSync(
         ReplicationCheckpoint checkpoint
     ) {
         // Update latest uploaded segment files name in segment tracker
-        segmentTracker.setLatestUploadedFiles(latestFileNameSizeOnLocalMap.keySet());
+        segmentTracker.setLatestUploadedFiles(segmentTracker.getLatestLocalFileNameLengthMap().keySet());
```

Comment:
Can there be a case where there can be a delay in updation of local files in segment tracker? mostly RemoteSegmentTrackerListener would finish before upload has completed but can there be a case of delay?

Reply:
No, the order is maintained. Code - https://github.com/opensearch-project/OpenSearch/pull/8758/files#diff-acf4481a07fc77c1b5d6eaca20a6a1ce1b171615c4349b67bd23b2142a2f0ccbR3677-R3683